src/algebra/regular.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2021 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import algebra.group
+
+/-!
+# Regular elements
+
+We introduce left-regular, right-regular and regular elements.  The final goal is to develop
+part of the API to prove, eventually, results about non-zero-divisors.
+
+-/
+variables {R : Type*}
+
+/-- A regular element is an element `c` such that multiplication by `c` both on the left and
+on the right is injective. -/
+structure is_regular [has_mul R] (c : R) : Prop :=
+(mul_left_inj : ∀ a b : R, c * a = c * b → a = b)
+(mul_right_inj : ∀ a b : R, a * c = b * c → a = b)
```

Comment:
Thoughts on instead using 
```suggestion
(mul_left_inj : function.injective ((*) c))
(mul_right_inj : function.injective (* c))
```

Which means you get the API of `injective` for free

Reply:
I am happy to go with this definition: as I mentioned, I can only see the immediate vicinity of the definition and have no idea what will make results easy/hard to prove later on.